util/sort.lua

```diff
@@ -72,34 +75,54 @@ end
 ---@param b Section
 ---@return boolean
 function sort.SortSectionsBySizeDescending(a, b)
-  return a:GetCellCount() > b:GetCellCount()
+  local aSize, bSize = a:GetCellCount(), b:GetCellCount()
+  if aSize ~= bSize then
+    return aSize > bSize
+  end
+  return a.title:GetText() < b.title:GetText()
 end
 
 ---@param a Section
 ---@param b Section
 ---@return boolean
 function sort.SortSectionsBySizeAscending(a, b)
-  return a:GetCellCount() < b:GetCellCount()
+  local aSize, bSize = a:GetCellCount(), b:GetCellCount()
+  if aSize ~= bSize then
+    return aSize < bSize
+  end
+  return a.title:GetText() < b.title:GetText()
 end
 
 ---@param a Item
 ---@param b Item
 ---@return boolean
 function sort.SortItemsByQualityThenAlpha(a, b)
   if invalidData(a.data, b.data) then return false end
-  if a.data.itemInfo.itemQuality == b.data.itemInfo.itemQuality then
+  if a.data.itemInfo.itemQuality ~= b.data.itemInfo.itemQuality then
+    return a.data.itemInfo.itemQuality > b.data.itemInfo.itemQuality
+  elseif a.data.itemInfo.currentItemLevel ~= b.data.itemInfo.currentItemLevel then
+    return a.data.itemInfo.currentItemLevel > b.data.itemInfo.currentItemLevel
+  elseif a.data.itemInfo.itemName ~= b.data.itemInfo.itemName then
     return a.data.itemInfo.itemName < b.data.itemInfo.itemName
+  elseif a.data.itemInfo.currentItemCount ~= b.data.itemInfo.currentItemCount then
+    return a.data.itemInfo.currentItemCount > b.data.itemInfo.currentItemCount
   end
-  return a.data.itemInfo.itemQuality > b.data.itemInfo.itemQuality
+  return a.data.itemInfo.itemGUID < b.data.itemInfo.itemGUID
 end
 
 ---@param a Item
 ---@param b Item
 ---@return boolean
 function sort.SortItemsByAlphaThenQuality(a, b)
   if invalidData(a.data, b.data) then return false end
-  if a.data.itemInfo.itemName == b.data.itemInfo.itemName then
+  if a.data.itemInfo.itemName ~= b.data.itemInfo.itemName then
+    return a.data.itemInfo.itemName < b.data.itemInfo.itemName
+  elseif a.data.itemInfo.itemQuality ~= b.data.itemInfo.itemQuality then
     return a.data.itemInfo.itemQuality > b.data.itemInfo.itemQuality
+  elseif a.data.itemInfo.currentItemLevel ~= b.data.itemInfo.currentItemLevel then
+    return a.data.itemInfo.currentItemLevel > b.data.itemInfo.currentItemLevel
+  elseif a.data.itemInfo.currentItemCount ~= b.data.itemInfo.currentItemCount then
+    return a.data.itemInfo.currentItemCount > b.data.itemInfo.currentItemCount
   end
-  return a.data.itemInfo.itemName < b.data.itemInfo.itemName
+  return a.data.itemInfo.itemGUID < b.data.itemInfo.itemGUID
```

Comment:
This is interesting, and I like this. Have you tested what this looks like more practically?

Reply:
It's just the last tiebreaker, something that is always unique. Only used when the item name, quality, (item level), and count are the same. In other words... by this point the items being sorted should be more or less indistinguishable. Maybe, using gear as an example, one has leech and one doesn't (or are just truly identical) - GUID sort as a final tiebreaker just makes sure they don't swap positions.